### PR TITLE
feat(clp-s): Add end to end search tests for clp-s.

### PR DESCRIPTION
**Pull request by gibber9809**
# Description
This PR adds end to end search tests for clp-s and associated boilerplate with initial tests inspired by #590 and #254. More tests can be added in the future to expand coverage.

This PR also touches test-clp_s-end-to-end in order to test single-file-archive support.

The basic flow for the search tests is:
1. Generate an archive based on test_search.jsonl
2. Run queries from queries_and_results one at a time and push results into a vector using VectorOutputHanlder
3. Check query results against expected results from queries_and_results vector

# Validation performed
* Validated that tests pass with expected results
* Validated that commented out test corresponds to a bug in Grep::process_raw_query

## Release Notes

- **New Features**
  - Enhanced search functionality with advanced query processing.
  - Added support for complex log search and filtering capabilities.
  - Introduced a new output handler for storing results in a vector.

- **Testing**
  - Expanded unit test coverage for search and compression operations.
  - Introduced new test cases for log processing and query validation.
  - Added a new test file for the `clp_s` search functionality.
  - Removed the automatic management of test output directories and files.

- **Performance**
  - Improved output handling with vector-based result storage.
  - Added more robust archive reading and processing mechanisms.

- **Infrastructure**
  - Updated build configuration to include new search and reducer components.

### CODE REVIEW

File: components/core/tests/test-clp_s-search.cpp [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+            {R"aa(NOT a: b)aa", {0}},
+            {R"aa(msg: "Msg 1: \"Abc123\"")aa", {1}},
+            {R"aa(msg: "Msg 2: 'Abc123'")aa", {2}},
+            {R"aa(msg: "Msg 3: \nAbc123")aa", {3}},
+            // CLP incorrectly generates no subqueries in Grep::process_raw_query for the following
+            // query, so we skip it for now.
+            //{R"aa(msg: "Msg 4: \\Abc123")aa", {4}}
+            {R"aa(msg: "Msg 5: \rAbc123")aa", {5}},
+            {R"aa(msg: "Msg 6: \tAbc123")aa", {6}},
+            {R"aa(msg: "*Abc123*")aa", {1, 2, 3, 5, 6}},
```
Comment (wraymo) [inline components/core/tests/test-clp_s-search.cpp:180]: I guess we'll also include more query shapes (e.g. `AND`, `?`) in the future?

**Review by wraymo — COMMENTED** [review]
A great PR! Looks good to me—I just have a few questions.

File: components/core/tests/test_log_files/test_search.jsonl [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{"idx": 0, "a": "clp string"}
+{"idx": 1, "msg": "Msg 1: \"Abc123\""}
+{"idx": 2, "msg": "Msg 2: 'Abc123'"}
+{"idx": 3, "msg": "Msg 3: \nAbc123"}
+{"idx": 4, "skip_msg": "Msg 4: \\Abc123"}
+{"idx": 5, "msg": "Msg 5: \rAbc123"}
+{"idx": 6, "msg": "Msg 6: \tAbc123"}
+{"idx": 7, "arr": [{"a": 999}, {"b": 1001}]}
+{"idx": 8, "arr": {"a": 999, "b": 1001}}
```
[the remark this answers is not in the document]
Comment (gibber9809) [reply components/core/tests/test_log_files/test_search.jsonl:9]: Yeah, just wanted to get all the boilerplate in + some key edge cases so that we have basic coverage and its easy to add more later.

Specifically I'll probably add a lot more cases to make sure I don't introduce regressions when I work on solving #588 since that fix will likely involve some tricky AST/ConvertToExists/Schema matching changes.

File: components/core/tests/LogSuppressor.hpp [hunk 3]
```diff
@@ -13,16 +13,17 @@ class LogSuppressor {
...
-    LogSuppressor& operator=(LogSuppressor&& other) noexcept = default;
-
     ~LogSuppressor() { spdlog::default_logger()->set_level(m_previous_logging_level); }
 
+    // Delete copy & move constructors and assignment operators
+    LogSuppressor(LogSuppressor const& other) = delete;
+    LogSuppressor(LogSuppressor&& other) = delete;
+    auto operator=(LogSuppressor const& other) -> LogSuppressor& = delete;
+    auto operator=(LogSuppressor&& other) -> LogSuppressor& = delete;
+
```
[the remark this answers is not in the document]
Comment (gibber9809) [reply components/core/tests/LogSuppressor.hpp:23]: We have some test cases that ensure illegal kql expressions fail to parse (and will do the same for sql). Inside of our KQL parser we log whenever we encounter a parsing error. We turn off logging just so that running those tests doesn't fill the terminal with error messages about failed parsing.

File: components/core/tests/test-clp_s-search.cpp (hunk 1 above)
Comment (gibber9809) [reply components/core/tests/test-clp_s-search.cpp:180]: Yep, just trying to get the boilerplate and basic coverage in for now. We can add more exhaustive tests as needed over time.

File: components/core/tests/LogSuppressor.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (wraymo) [reply components/core/tests/LogSuppressor.hpp:23]: Got it. Yeah that makes sense.

**Review by wraymo — APPROVED** [review]
